python/fledge/services/core/api/auth.py

```diff
@@ -680,6 +693,58 @@ async def enable_user(request):
         raise web.HTTPInternalServerError(reason=str(exc), body=json.dumps({"message": msg}))
     return web.json_response({'message': 'User with ID:<{}> has been {} successfully.'.format(int(user_id), _text)})
 
+@has_permission("admin")
+async def unblock_user(request):
+    """ Unblock the user got blocked due to multiple invalid log in attempts
+        :Example:
+            curl -H "authorization: <token>" -X PUT  http://localhost:8081/fledge/admin/{user_id}/unblock
+    """
+    if request.is_auth_optional:
+        _logger.warning(FORBIDDEN_MSG)
+        raise web.HTTPForbidden
+
+    user_id = request.match_info.get('user_id')
+
+    try:
+        from fledge.services.core import connect
+        storage_client = connect.get_storage_async()
+        await _unblock_user(user_id,storage_client)
+
+        # USRUB audit trail entry
+        audit = AuditLogger(storage_client)
+        await audit.information('USRUB', {'user_id': int(user_id),
+                        "message": "User with ID:<{}> has been unblocked.".format(user_id)})
+
+    except ValueError as err:
+        msg = str(err)
+        raise web.HTTPBadRequest(reason=str(err), body=json.dumps({"message": msg}))
+    except User.DoesNotExist:
+        msg = "User with ID:<{}> does not exist.".format(int(user_id))
+        raise web.HTTPNotFound(reason=msg, body=json.dumps({"message": msg}))
+    except Exception as exc:
+        msg = str(exc)
+        _logger.error(exc, "Failed to unblock user ID:<{}>.".format(user_id))
+        raise web.HTTPInternalServerError(reason=str(exc), body=json.dumps({"message": msg}))
+    return web.json_response({'message': 'User with ID:<{}> has been unblocked successfully.'.format(int(user_id))})
+
+
+async def _unblock_user(user_id, storage_client):
+    """ implementation for unblock user
+    """
+
+    from fledge.common.storage_client.payload_builder import PayloadBuilder
+
+    payload = PayloadBuilder().SELECT("id", "uname", "role_id", "block_until", "failed_attempts").WHERE(
```

Comment:
We do not need to fetch all of the columns here; Simple with one column and check if storage returns any row then move else raise the condition